support-models/src/main/scala/com/gu/support/redemptions/RedemptionCode.scala

```diff
@@ -6,12 +6,14 @@ import com.gu.support.encoding.Codec
 import io.circe.{Decoder, Encoder}
 
 object RedemptionCode {
+  val length = 13
+
   def apply(value: String): Either[String, RedemptionCode] = {
     val upper = value.toUpperCase(Locale.UK)
     // make sure no one can inject anything bad
     val validChars = List('A' -> 'Z', '0' -> '9', '-' -> '-')
     val validCharsSet = validChars.flatMap { case (from, to) => (from to to) }.toSet
-    if (upper.forall(validCharsSet.contains))
+    if (upper.length == length && upper.forall(validCharsSet.contains))
       Right(new RedemptionCode(upper))
     else
       Left(s"redemption code must only include A-Z, 0-9 and '-'")
```

Comment:
```suggestion
      Left(s"redemption code must only include A-Z, 0-9 and '-' and be 13 characters long")
```

Reply:
I'm removing the server side validlation for now because it would mean changing test data in Dynamo. I have a follow up PR coming that will do that anyway so I'll roll it into that.